File: buffer/gpu/src/jvmMain/java/com/wsr/gpu/elementwise/operation/JTimes.java
package com.wsr.gpu.elementwise.operation;

public class JTimes {
    public native void timesD0ToD1(float x, long y, long result, long runtime);

    public native void timesD1ToD0(long x, float y, long result, long runtime);

    public native void timesD1ToD1(long x, long y, long result, long runtime);

    public native void timesD1ToD2(long x, long y, int yi, int yj, int axis, long result, long runtime);

    public native void timesD1ToD3(long x, long y, int yi, int yj, int yk, int axis, long result, long runtime);

    public native void timesD2ToD1(long x, int xi, int xj, long y, int axis, long result, long runtime);

    public native void timesD2ToD3(
            long x,
            int xi,
            int xj,
            long y,
            int yi,
            int yj,
            int yk,
            int axis1,
            int axis2,
            long result,
            long runtime
    );

    public native void timesD3ToD1(long x, int xi, int xj, int xk, long y, int axis, long result, long runtime);

    public native void timesD3ToD2(
            long x,
            int xi,
            int xj,
            int xk,
            long y,
            int yi,
            int yj,
            int axis1,
            int axis2,
            long result,
            long runtime
    );

    public native void timesD3ToD4(
            long x,
            int xi,
            int xj,
            int xk,
            long y,
            int yi,
            int yj,
            int yk,
            int yl,
            int axis1,
            int axis2,
            int axis3,
            long result,
            long runtime
    );

    public native void timesD4ToD1(long x, int xi, int xj, int xk, int xl, long y, int axis, long result, long runtime);

    public native void timesD4ToD2(
            long x,
            int xi,
            int xj,
            int xk,
            int xl,
            long y,
            int yi,
            int yj,
            int axis1,
            int axis2,
            long result,
            long runtime
    );

    public native void timesD4ToD3(
            long x,
            int xi,
            int xj,
            int xk,
            int xl,
            long y,
            int yi,
            int yj,
            int yk,
            int axis1,
            int axis2,
            int axis3,
            long result,
            long runtime
    );
}

File: buffer/cpu/src/jvmAndroidMain/kotlin/com/wsr/knist/cpu/elementwise/operation/times/JTimes.kt
package com.wsr.knist.cpu.elementwise.operation.times


object JTimes {
    external fun timesD0ToD1(x: Float, y: Long, result: Long)

    external fun timesD1ToD0(x: Long, y: Float, result: Long)
    external fun timesD1ToD1(x: Long, y: Long, result: Long)
    external fun timesD1ToD2(x: Long, y: Long, yi: Int, yj: Int, axis: Int, result: Long)
    external fun timesD1ToD3(x: Long, y: Long, yi: Int, yj: Int, yk: Int, axis: Int, result: Long)

    external fun timesD2ToD1(x: Long, xi: Int, xj: Int, y: Long, axis: Int, result: Long)
    external fun timesD2ToD3(
        x: Long,
        xi: Int,
        xj: Int,
        y: Long,
        yi: Int,
        yj: Int,
        yk: Int,
        axis1: Int,
        axis2: Int,
        result: Long,
    )

    external fun timesD3ToD1(x: Long, xi: Int, xj: Int, xk: Int, y: Long, axis: Int, result: Long)
    external fun timesD3ToD2(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        y: Long,
        yi: Int,
        yj: Int,
        axis1: Int,
        axis2: Int,
        result: Long,
    )
    external fun timesD3ToD4(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        y: Long,
        yi: Int,
        yj: Int,
        yk: Int,
        yl: Int,
        axis1: Int,
        axis2: Int,
        axis3: Int,
        result: Long,
    )

    external fun timesD4ToD1(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        xl: Int,
        y: Long,
        axis: Int,
        result: Long,
    )
    external fun timesD4ToD2(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        xl: Int,
        y: Long,
        yi: Int,
        yj: Int,
        axis1: Int,
        axis2: Int,
        result: Long,
    )
    external fun timesD4ToD3(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        xl: Int,
        y: Long,
        yi: Int,
        yj: Int,
        yk: Int,
        axis1: Int,
        axis2: Int,
        axis3: Int,
        result: Long,
    )
}
